presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -628,20 +628,24 @@ private PlanBuilder sort(PlanBuilder subPlan, List<SortItem> orderBy, Optional<S
         Iterator<SortItem> sortItems = orderBy.iterator();
 
         ImmutableList.Builder<Symbol> orderBySymbols = ImmutableList.builder();
-        ImmutableMap.Builder<Symbol, SortOrder> orderings = ImmutableMap.builder();
+        Map<Symbol, SortOrder> orderings = new HashMap<Symbol, SortOrder>();
         for (FieldOrExpression fieldOrExpression : orderByExpressions) {
             Symbol symbol = subPlan.translate(fieldOrExpression);
-            orderBySymbols.add(symbol);
+            if (orderings.containsKey(symbol)) {
+                sortItems.next();
```

Comment:
Instead of calling next() in the two branches (which makes the code a little more error prone), assign sortItems.next() to a variable before the `if` block and use it below. 

Also, invert the `if` condition. I think the code will read more natural:

```
for (...) {
    ...
    if (!orderings.contains(...)) {
       ... add to map, etc
    }
}